Update Orleans and Aspire integration documentation (#54201)
dotnet/docs: docs/orleans/host/aspire-integration.md

:::zone target="docs" pivot="orleans-8-0,orleans-9-0,orleans-10-0"

> [Aspire](https://aspire.dev/integrations/frameworks/orleans) provides a streamlined approach to building cloud-native applications with built-in support for Orleans. Starting with Orleans 8.0, you can use Aspire to orchestrate your Orleans cluster, manage backing resources (like Redis or Azure Storage), and automatically configure service discovery, observability, and health checks.

Reviewer: The original link makes more sense to me here.

Reviewer 2: I thought about that too. To me, someone is here because they probably know what aspire is and are more interested in integrating the two techs. So a link to the integration would make sense over a "what is aspire"

But I can go either way.

> [Aspire](https://aspire.dev/integrations/frameworks/orleans) provides a streamlined approach to building cloud-native applications with built-in support for Orleans. Starting with Orleans 8.0, you can use Aspire to orchestrate your Orleans cluster, manage backing resources (like Redis or Azure Storage), and automatically configure service discovery, observability, and health checks.

Reviewer: ```suggestion
[Aspire](https://aspire.dev/get-started/what-is-aspire/) provides a streamlined approach to building cloud-native applications with built-in support for Orleans. Starting with Orleans 8.0, you can use Aspire to orchestrate your Orleans cluster, manage backing resources (like Redis or Azure Storage), and automatically configure service discovery, observability, and health checks.
```

Before using Orleans with Aspire, ensure you have:

- [.NET 8.0 SDK](https://dotnet.microsoft.com/download/dotnet/8.0) or later

> - [Aspire cli](https://aspire.dev/get-started/install-cli/)

Reviewer: ```suggestion
- [Aspire CLI](https://aspire.dev/get-started/install-cli/)
```

- [.NET 8.0 SDK](https://dotnet.microsoft.com/download/dotnet/8.0) or later
- [Aspire CLI](https://aspire.dev/get-started/install-cli/)
- An IDE with Aspire support, such as one of the following:

>   - Visual Studio 2022 17.9+

Reviewer: ```suggestion
  - Visual Studio 2022 version 17.9+
```

  - Visual Studio Code with C# Dev Kit and the [`microsoft-aspire.aspire-vscode`](https://marketplace.visualstudio.com/items?itemName=microsoft-aspire.aspire-vscode) extension.
  - JetBrains Rider

> - Familiarize yourself with the Aspire Orleans integration packhttps://aspire.dev/integrations/frameworks/orleans

Reviewer: packhttps?

Reviewer 2: Strange.. I don't know how the link info got lost...
```suggestion
- Familiarize yourself with the [Aspire Orleans integration pack](https://aspire.dev/integrations/frameworks/orleans)
```

## AppHost extension methods reference

> The [`Aspire.Hosting.Orleans` package](https://aspire.dev/integrations/frameworks/orleans/) provides these extension methods:

Reviewer: ```suggestion
The [`Aspire.Hosting.Orleans` package](https://www.nuget.org/packages/Aspire.Hosting.Orleans) provides these extension methods:
```

:::zone target="docs" pivot="orleans-7-0"

Aspire integration was introduced in Orleans 8.0. For Orleans 7.0, you can still deploy to Aspire-orchestrated environments, but the dedicated `Aspire.Hosting.Orleans` package and its extension methods are not available.

Consider upgrading to Orleans 8.0 or later to take advantage of the Aspire integration features.

:::zone-end

:::zone target="docs" pivot="orleans-3-x"

> Aspire integration is available in Orleans 8.0 and later. Orleans 3.x does not support .NET Aspire.

Reviewer: ```suggestion
Aspire integration is available in Orleans 8.0 and later. Orleans 3.x does not support Aspire.
```
